UX: infer partial crop params and add progress bar

DURUII opened the pull request:
### Background
Running with only `crop_y` appeared valid but was silently ignored with no feedback. This caused confusion and prompted this change. For example, 

<img width="935" height="766" alt="image" src="https://github.com/user-attachments/assets/1ddab3a5-5bbe-4d89-a348-69453f88030d" />

'那早晚有一天会结束 CHERY' should not appear here, since the user have carefully set the `crop_y`.

### Changes
- **Partial crop inference**: default missing `x/y` to 0, `width/height` to remaining frame; clamp to bounds; fallback to lower 30% if zero-area.
- **Clear warnings**:
  - Incomplete crop → show inferred region.
  - `use_fullframe=True` with crop → ignored with warning.
  - No crop with `use_fullframe=False` → default to lower 30% with notice.
- **Progress feedback**: add `tqdm` progress; updates once per processed frame.

<img width="1434" height="448" alt="image" src="https://github.com/user-attachments/assets/b0fc8163-e4a7-42c8-87bc-ea2aba25bdba" />

### Why
- **User hint for partial inputs**: prevents silent misconfiguration.
- **Ease waiting anxiety**: minimal, non-noisy progress feedback.

### Notes
- No public API changes.
- Files: `videocr/video.py`.
- `tqdm` is optional.

DURUII committed:
feat(video): hint for partial crop inputs; one-line progress bar

- Support partial crop inputs and Improve messaging: Default missing x/y to 0, width/height to remaining frame
- Add optional one-line tqdm progress to ease waiting anxiety
- No public API changes; tqdm remains an optional dependency

devmaxxing commented on videocr/video.py:
```diff
@@ -68,26 +71,68 @@ def run_ocr(self, use_gpu: bool, lang: str, time_start: str, time_end: str,
-            crop_x_end = crop_x + crop_width
-            crop_y_end = crop_y + crop_height
+
+        # improved cropping behavior with informative warnings
```
remove these kinds of comments

devmaxxing reviewed (changes requested):
Looks ok. just remove some of the excessive comments. 
Also, can you provide manual test results covering your changes?

DURUII commented:
> Looks ok. just remove some of the excessive comments. Also, can you provide manual test results covering your changes?

I’ve addressed the comments:

* Removed excessive comments.
* Added a short manual test report below.

### What changed (recap)

* **Partial crop inference**: if only `crop_x/crop_y/crop_width/crop_height` are partially provided, the missing ones are inferred (x/y→0, width/height→remaining frame) and clamped to bounds. If the region collapses to zero, we **fallback to the lower 30%** and print a notice.
* **Clear warnings**:

  * Incomplete crop → print the inferred crop rectangle.
  * `use_fullframe=True` with any crop → crop is ignored, warn once.
  * No crop while `use_fullframe=False` → default to lower 30% with notice.
* **Progress feedback**: optional `tqdm` updates once per processed frame; completely silent if `tqdm` is unavailable or `pbar=None`.
* **No public API changes**. Touches only `videocr/video.py`. `tqdm` is optional.

### Manual tests

✅ **Case A (only `crop_y/x`)**: warning shows inferred crop; text that previously leaked from upper region is no longer OCR’d.

<img width="1487" height="342" alt="image" src="https://github.com/user-attachments/assets/b4de46ff-c585-4690-b865-5912e1f2304a" />

✅ **Case B (`use_fullframe=True` + crop given)**: single warning that crop is ignored; subtitles identical to full-frame run.

<img width="1466" height="299" alt="image" src="https://github.com/user-attachments/assets/a720cb1f-d599-4f37-9c76-f71a2f5b49b1" />

✅ **Case C (no crop + `use_fullframe=False`)**: notice about defaulting to lower 30%; OCR limited to bottom band (expected for typical caption area).

<img width="1710" height="364" alt="Screenshot 2025-08-17 at 11 25 37 AM" src="https://github.com/user-attachments/assets/f6ce9b3f-c591-44fe-8cf9-a721961e7e9d" />

✅ **Edge/robustness checks**: Provide a crop that would be out of bounds or zero-area (e.g., `crop_height=0` or `crop_y` ≥ frame height) → you should see a **clamped** region or a **fallback to lower 30%** with a notice, not a silent pass.

<img width="1484" height="528" alt="image" src="https://github.com/user-attachments/assets/5e042922-61d1-4d26-b304-51c00ad2a23f" />

devmaxxing commented:
Nice.
谢谢兄弟
